web/src/server_events_dispatch.js

```diff
@@ -546,20 +547,22 @@ export function dispatch_normal_event(event) {
                         const is_narrowed_to_stream = narrow_state.is_for_stream_id(
                             stream.stream_id,
                         );
+                        stream_data.delete_sub(stream.stream_id);
                         if (is_narrowed_to_stream) {
                             message_lists.current.update_trailing_bookend();
                         }
-                        stream_data.delete_sub(stream.stream_id);
                         stream_settings_ui.remove_stream(stream.stream_id);
+                        message_view_header.maybe_rerender_title_area_for_stream(stream);
+                        message_live_update.rerender_messages_view();
                         if (was_subscribed) {
                             stream_list.remove_sidebar_row(stream.stream_id);
                             if (stream.stream_id === compose_state.selected_recipient_id) {
                                 compose_state.set_selected_recipient_id("");
                                 compose_recipient.on_compose_select_recipient_update();
                             }
                         }
-                        settings_streams.update_default_streams_table();
                         stream_data.remove_default_stream(stream.stream_id);
+                        settings_streams.update_default_streams_table();
```

Comment:
I think the logical order this function is wrong.

`delete_sub` must be called before `update_trailing_bookend`; otherwise, an incorrect trailing end will be presented.

If a stream is opened and subsequently deactivated, the following trailing end used to be visible:
![image](https://github.com/zulip/zulip/assets/98596642/9b2f4a4c-f505-4319-bc6b-76b9107e753c)

That's why I was surprised when I discovered the existence of a specific trailing end for deactivated streams. Despite numerous interactions with deactivating and deactivated streams, I never saw it, partly due to this incorrect order.

Likewise `remove_default_stream` should be called before `update_default_streams_table` since `update_default_streams_tables` invokes `build_default_stream_table`  which uses `default_stream_ids` before it is updated.

Reply:
Good finds!

Reply:
Done.